daemon/hostdaemon.go

```diff
@@ -82,18 +112,96 @@ func (d *HostDaemon) connectWithRetry() {
 	conn, err := grpc.Dial(fmt.Sprintf("%s:%d", d.addr, d.port), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithDefaultServiceConfig(retryPolicy))
 	if err != nil {
 		d.log.Error(err, "did not connect")
+		return err
 	}
 	d.conn = conn
 	d.client = pb.NewBridgePortServiceClient(conn)
+	return nil
+}
+
+func (d *HostDaemon) cniCmdAddHandler(req *cnitypes.PodRequest) (*cni100.Result, error) {
+	pf := 0
+	vf := req.CNIConf.VFID
+	mac := req.CNIConf.RuntimeConfig.Mac
+	d.log.Info("addHandler", "CNIConf", req.CNIConf)
+	d.log.Info("addHandler", "vlan", req.CNIConf.Vlan)
+	vlan := *req.CNIConf.Vlan
+	d.log.Info("addHandler", "pf", pf, "vf", vf, "mac", mac, "vlan", vlan)
+	_, err := d.CreateBridgePort(pf, vf, vlan, mac)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to call CreateBridgePort: %v", err)
+	}
+	d.log.Info("addHandler CreateBridgePort succeeded")
+
+	res, err := d.sm.CmdAdd(req)
+	if err != nil {
+		return nil, errors.New("SRIOV manager failed in add handler")
+	}
+	d.log.Info("addHandler d.sm.CmdAdd succeeded")
+	return res, nil
+}
+
+func (d *HostDaemon) cniCmdDelHandler(req *cnitypes.PodRequest) (*cni100.Result, error) {
+	pf := 0
+	vf := req.CNIConf.VFID
+	mac := req.CNIConf.MAC
+	vlan := 7//*req.CNIConf.Vlan
```

Comment:
We should just use req.CNIConf.Vlan here. I don't see why this won't work.

Reply:
thanks for catching this commented out code.